src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -179,99 +165,104 @@ class DataModelInstanceRelation(config: RelationConfig, modelExternalId: String)
+            })
+        }
+        .collect { case (a, Some(value)) => a -> value }
+        .toMap
+
+      DataModelInstanceCreate(modelExternalId, properties = Some(propertyValues))
     }
-    parseRow
+    parseRow(indexedPropertyList)
   }
 }
 
 object DataModelInstanceRelation {
-  private def propertyNotNullableMessage(propertyType: String) =
-    s"Property of $propertyType type is not nullable."
   private def unknownPropertyTypeMessage(a: Any) = s"Unknown property type $a."
 
   private def notValidPropertyTypeMessage(a: Any, propertyType: String) =
     s"$a is not a valid $propertyType"
 
+  private def propertyNotNullableMessage(propertyType: String) =
+    s"Property of $propertyType type is not nullable."
+
   //scalastyle:off cyclomatic.complexity
-  private def parseValue(value: Any): Json = value match {
-    case x: Double => jsonFromDouble(x)
-    case x: Int => jsonFromDouble(x.toDouble)
-    case x: Float => jsonFromDouble(x.toDouble)
-    case x: Long => jsonFromDouble(x.toDouble)
-    case x: java.math.BigDecimal => jsonFromDouble(x.doubleValue)
-    case x: java.math.BigInteger => jsonFromDouble(x.doubleValue)
-    case x: String => Json.fromString(x)
-    case x: Boolean => Json.fromBoolean(x)
-    case x: Array[Any] =>
-      Json.fromValues(x.map(parseValue))
-    case null => Json.Null // scalastyle:off null
+  private def parsePropertyValue(value: Any): PropertyType = value match {
+    case x: Double => Float64Property(x)
+    case x: Int => Int32Property(x)
+    case x: Float => Float32Property(x)
+    case x: Long => Int64Property(x)
+    case x: java.math.BigDecimal => Float64Property(x.doubleValue())
+    case x: java.math.BigInteger => Int64Property(x.longValue())
+    case x: String => StringProperty(x)
+    case x: Boolean => BooleanProperty(x)
+    case x: Array[Double] => ArrayProperty(x.toVector.map(Float64Property))
+    case x: Array[Int] => ArrayProperty(x.toVector.map(Int32Property))
+    case x: Array[Float] => ArrayProperty(x.toVector.map(Float32Property))
+    case x: Array[Long] => ArrayProperty(x.toVector.map(Int64Property))
+    case x: Array[String] => ArrayProperty(x.toVector.map(StringProperty))
+    case x: Array[Boolean] => ArrayProperty(x.toVector.map(BooleanProperty))
+    case x: Array[java.math.BigDecimal] =>
+      ArrayProperty(x.toVector.map(i => Float64Property(i.doubleValue())))
+    case x: Array[java.math.BigInteger] =>
+      ArrayProperty(x.toVector.map(i => Int64Property(i.longValue())))
+    case x: LocalDate => DateProperty(x)
+    case x: java.sql.Date => DateProperty(x.toLocalDate)
+    case x: LocalDateTime => DateProperty(x.toLocalDate)
+    case x: Instant => TimeStampProperty(x.atZone(ZoneId.systemDefault()))
+    case x: java.sql.Timestamp =>
+      TimeStampProperty(x.toInstant.atZone(ZoneId.systemDefault()))
+    case x => throw new CdfSparkException(s"Cannot parse the value with udentified type: $x")
```

Comment:
Here's the typo, in "udentified". I don't know what happened to my previous comment... but my suggestion was `s"Unsupported value ${x.toString} of type ${x.getClass.getName}"`